hdr/sqlite_modern_cpp.h

```diff
@@ -848,6 +846,29 @@ namespace sqlite {
 	 inline void store_result_in_db(sqlite3_context* db, const std::u16string& val) {
 		 sqlite3_result_text16(db, val.data(), -1, SQLITE_TRANSIENT);
 	}
+
+	// Other integer types
+	 template<class Integral, class = typename std::enable_if<std::is_integral<Integral>::value>::type>
+	 inline database_binder& operator <<(database_binder& db, const Integral& val) {
+		return db << static_cast<sqlite3_int64>(val);
```

Comment:
Actually `int` will still be handled by the `int` overload. Your version would lead to a failure for `unsigned int`, if the value doesn't fit into an `int`. Additionally this is the original implementation of `sqlite3_bind_int`:

```c
SQLITE_API int sqlite3_bind_int(sqlite3_stmt *p, int i, int iValue){
  return sqlite3_bind_int64(p, i, (i64)iValue);
}
```
So we don't really have to care about calling it for small values anyway.